ddtrace/tracer/payload_test.go

```diff
@@ -95,7 +93,6 @@ func benchmarkPayloadThroughput(count int) func(*testing.B) {
 		b.ReportAllocs()
 		b.ResetTimer()
 		for i := 0; i < b.N; i++ {
-			p.reset()
```

Comment:
Doesn't this invalidate the benchmark? The payload will never get cleared. Maybe we need to say `p := newPayload()` here instead?

Reply:
I guess I didn't pay enough attention here, I misunderstood this benchmark. I added back the reset, scoped solely to the benchmark. WDYT?

Reply:
LGTM.